### ISSUE TITLE
x instanceof imported.SumType is not working (false positive TS2339)

### ISSUE THREAD

**Issue by serkonda7** [microsoft/TypeScript#52799]
# Bug Report

### 🔎 Search Terms
- TS2339

### 🕗 Version & Regression Information
- 4.9.5

### ⏯ Playground Link
> Note: the error is not triggered in the workbench!
Bug workbench: https://www.typescriptlang.org/dev/bug-workbench/?ts=4.9.5#code/PTAEAEDMEsBsFMB2BDAtvAXKA2sgdAC4DOAugFAECeADvKACo3wAi8AxrKALygCCs0ZEUa1QAH1ABlAK6oR8MhSZSCqAt1ABRAB7UATpNXqJ81h0UchRLboNGA3gF8LsK3wFD5Tl25lym3mTwugD2eur2KmoANDb6hjHugsJMsX7ysabsnM6KIBAwCCjoWNgARoSkZNCo1GHqAFSgQs1E6pB6IaigAOR4wMg9ipDSiGwE0CGIoJAhIQAUbWpYQgR4CQQAlPZkAJD58HqdeqDQ1gAWh5h70JCgi0aniG3IY-Ahd6t4WRzbzrn5ADqYQA1tYYIg6Jc9NcRmMJlNQGVkHp5gQACbZFZtb5MMywP5AA

### 💻 Code
```ts
// a.ts
type TypeDecl = AliasType | SumType

type Stmt = ExprStmt | TypeDecl

class ExprStmt{}

class AliasType{}

class SumType{}

export { Stmt, ExprStmt, AliasType, SumType, TypeDecl }
```

```ts
// b.ts

import * as ast from './a'

function foo(stmt: ast.Stmt){
	// error is here:
	if (stmt instanceof ast.TypeDecl){}
}

// Works fine here:
function bar(tdecl: ast.TypeDecl){}
```

### 🙁 Actual behavior
I get the error in the line where I check if my variable is the `TypeDecl` variant of the `Stmt` sumtype.
However `TypeDecl` is a sumtype on it's own.
```
Property 'TypeDecl' does not exist on type 'typeof import("/home/john/Documents/local_repos/compiler_in_ts/sample/a")'.
```

### 🙂 Expected behavior
Do not give an error as it actually exists. The annotation below the error is unproblematic too.

**Comment by MartinJohns** [comment]
`instanceof` does not work with type aliases. Type aliases are a TypeScript only construct and get erased during compilation, but `instanceof` is a JavaScript operator that checks for the prototype-chain.

Or differently said: `instanceof` requires a **value**, but your `TypeDecl` is only a **type**.

**Comment by serkonda7** [comment]
How to perform this check then?

**Comment by fatcerberus** [comment]
However you would do it in JS, which will depend on the specific type(s) involved.  TypeScript types are purely a compile-time construct.

**Comment by fatcerberus** [comment]
> or even generate some code like …

TypeScript intentionally doesn’t generate code for type-level operations.  That the types only exist at check-time is 100% by-design.  Refer to the [TypeScript Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals) and in particular **Non-Goal** # 5:

> [Do not] Add or rely on run-time type information in programs, or emit different code based on the results of the type system. Instead, encourage programming patterns that do not require run-time metadata.

I agree about the error message though, usually TS tells you when you’ve incorrectly used a type in value positions; it’s weird that it doesn’t do so here.
